tests/integration/test_balancer.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+from subprocess import PIPE, CalledProcessError, check_output
+
+import pytest
+from pytest_operator.plugin import OpsTest
+
+from .helpers import (
+    APP_NAME,
+    ZK_NAME,
+)
+
+logger = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.balancer
+
+
+def balancer_is_running(model_full_name: str | None) -> bool:
+    try:
+        check_output(
+            f"JUJU_MODEL={model_full_name} juju ssh kafka/0 sudo -i 'curl http://localhost:9090/kafkacruisecontrol/state'",
+            stderr=PIPE,
+            shell=True,
+            universal_newlines=True,
+        )
+    except CalledProcessError:
+        return False
+    return True
```

Comment:
**suggestion:** I think it's fine to just raise `CalledProcessError` here. If the test fails, we'd want to get the actual traceback, rather than it being hidden in the `AssertionError` that would have been raised.

We could do:
```python
assert balancer_is_running()
```

Reply:
Yes, that makes perfect sense. I first wrote this helper having the `assert not balancer_is_running()` statement in mind, but I fully agree that keeping the traceback is way better. Changed in 97f9a44f678c30e812875aa4bca0dac2a80e6b9d